app/src/main/java/de/westnordost/streetcomplete/quests/address/AddHousenumberForm.kt

```diff
@@ -240,27 +270,38 @@ class AddHousenumberForm : AbstractQuestFormAnswerFragment<HousenumberAnswer>()
         }
     }
 
-    private fun createAnswer(): HousenumberAnswer? =
-        if (houseNameInput != null) {
-            houseNameInput?.nonEmptyInput?.let { HouseName(it) }
-        }
-        else if (conscriptionNumberInput != null && streetNumberInput != null) {
-            conscriptionNumberInput?.nonEmptyInput?.let { conscriptionNumber ->
-                val streetNumber = streetNumberInput?.nonEmptyInput // streetNumber is optional
-                ConscriptionNumber(conscriptionNumber, streetNumber)
-            }
+    override fun isRejectingClose(): Boolean {
+        val houseName = houseNameInput?.nonEmptyInput
+        val houseNumber = houseNumberInput?.nonEmptyInput
+        val conscriptionNumber = conscriptionNumberInput?.nonEmptyInput
+        val streetNumber = streetNumberInput?.nonEmptyInput
+        val blockNumber = blockNumberInput?.nonEmptyInput
+
+        if(houseName != null || houseNumber != null) {
+            return true
```

Comment:
So, user can answer "it has both number and name" but then don't fill out both the fields and still OK it?

I think this is confusing.

That means that the "OK" button will appear as soon as he inputs the first letter for the housename.

The "OK" button should only appear after there is one character at least in **both** input fields, since this is what the user stated that he wanted to input.

Reply:
That is in `isRejectingClose` - so it is checking whether user should confirm discarding partially filled info.

`isFormComplete` is higher.

Reply:
I had to put the comment somewhere. But If I misunderstood the behavior from reading the code and it is already the way I described, then you can simply reoslve this conversation.

Reply:
oh right, it was actually broken